sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/TimestampPolicy.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.rabbitmq;
+
+import java.io.Serializable;
+import java.util.Date;
+import org.apache.beam.sdk.io.UnboundedSource;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.joda.time.Instant;
+
+/**
+ * A timestamp policy to assign event time for messages from a RabbitMq queue and watermark for it.
+ *
+ * <p>Inspired by a similar concept in KafkaIO
+ */
+public abstract class TimestampPolicy {
+
+  /**
+   * A timestamp extractor compatible with the <a
+   * href="https://github.com/rabbitmq/rabbitmq-message-timestamp">RabbitMQ Message Timestamp
+   * plugin</a>, which first looks for header {@code timestamp_in_ms} then falls back to amqp
+   * message property {@code timestamp}
+   */
+  public static SerializableFunction<RabbitMqMessage, Instant>
```

Comment:
Extracted this extremely useful, common instance of extracting a timestamp. Seems useful.